### PR TITLE
[REFERENT SPACE] Add co-referent management to the referent organizational chart

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/ReferentController.php [hunk 1]
```diff
@@ -506,13 +507,30 @@ public function jecouteSurveyStatsDownloadAction(Survey $survey, StatisticsExpor
 
     /**
      * @Route("/mon-equipe", name="app_referent_organizational_chart")
+     * @Method({"GET", "POST"})
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ReferentController.php]: Can you use ` methods={"GET|POST"}` in the `@Route` annotations instead ?

File: src/Controller/EnMarche/ReferentController.php [hunk 2]
```diff
@@ -506,13 +507,30 @@ public function jecouteSurveyStatsDownloadAction(Survey $survey, StatisticsExpor
...
+        $form = $this
+            ->createForm(PotentialCoReferentsType::class, ['referentPersonLinks' => $personalLinks])
+            ->handleRequest($request)
+        ;
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            foreach ($personalLinks as $personalLink) {
+                $personalLink->updateCoReferentRole($this->getUser());
+            }
+            $this->getDoctrine()->getManager()->flush();
```
Comment (thomasbisignani) [inline src/Controller/EnMarche/ReferentController.php]: What do you think to inject the `ObjectManager` ?

File: translations/messages.fr.yml [hunk 3]
```diff
@@ -266,6 +266,9 @@ referent.adherent.all: Tous
 referent.adherent.subscribed: Abonné(e)
 referent.adherent.unsubscribed: Non-abonné(e)
 referent.status.email: Statut (e-mail)
+referent.checkbox.co_referent: Donner un accès à l'onglet <b>Adhérents</b> en le nommant co-référent
+referent.save: Enrégistrer
```
Comment (thomasbisignani) [inline translations/messages.fr.yml]: `Enrégistrer` => `Enregistrer`

File: src/Controller/EnMarche/ReferentController.php (hunk 1 above)
Comment (JeremieSamson) [reply src/Controller/EnMarche/ReferentController.php]: Will not be a problem anymore with #3556 :+1:

File: src/Entity/ReferentTeam.php [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+     */
+    private $referent;
+
+    public function __construct(Adherent $adherent, Adherent $referent)
+    {
+        $this->adherent = $adherent;
+        $this->referent = $referent;
+    }
+
+    public function getId()
```
Comment (Remg) [inline src/Entity/ReferentTeam.php]: missing return type

File: src/Entity/ReferentTeam.php [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        return $this->id;
+    }
+
+    public function getAdherent(): ?Adherent
+    {
+        return $this->adherent;
+    }
+
+    public function setAdherent(Adherent $adherent = null)
```
Comment (Remg) [inline src/Entity/ReferentTeam.php]: missing return type

File: src/Entity/ReferentTeamMember.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="referent_team_member")
```
Comment (ottaviano) [inline src/Entity/ReferentTeamMember.php]: @table is optional

**Pull request merged**